client/my-sites/guided-transfer/host-credentials-page/index.jsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Bluehost from './bluehost';
+import SiteGround from './siteground';
+import SectionHeader from 'components/section-header';
+
+class HostCredentialsPage extends Component {
+	static propTypes = {
+		hostSlug: PropTypes.string.isRequired,
+	};
+
+	state = { fieldValues: {} };
```

Comment:
I'd lean toward agreeing that keeping this in the component's state makes the most sense given the component hierarchy, for all of the reasons @jordwest highlighted in his last comment. Shared refs between components seems particularly troublesome. I do think there's a case for uncontrolled forms, especially in cases where the contents of the form may be discarded. In #5600, I changed the [`<EditorMediaAdvanced />`](https://github.com/Automattic/wp-calypso/pull/5600/files#diff-1) component from being a Redux-bound form to an uncontrolled form with `refs` because it was much easier to implement than trying to reset state when the form was shown or hidden.

Whether Redux state would be the most appropriate place for this... I've not been convinced strongly one way or the other yet. I think there's significant boilerplate in binding a component's state to Redux, and for simple cases like this, that cost usually doesn't seem appropriate. As @jordwest highlighted though, it'd be much easier to migrate state to Redux than an uncontrolled form.